### DISCUSSION TITLE
[Feature] improve logging to show real filenames

### DISCUSSION THREAD

**Discussion by SlavikCA** [immich-app/immich#5808] [Feature Request]
### The feature

I'm using Immich in Docker 1.91.3 with external library.

When I started the scan process, I'm getting ffmpeg errors like this:

> 
> [Nest] 7  - 12/17/2023, 10:30:52 PM   ERROR [JobService] Error: ffprobe exited with code 1
> ffprobe version 6.0-Jellyfin Copyright (c) 2007-2023 the FFmpeg developers
>   built with gcc 12 (Debian 12.2.0-14)
>   configuration: --prefix=/usr/lib/jellyfin-ffmpeg --target-os=linux --extra-version=Jellyfin --disable-doc --disable-ffplay --disable-ptx-compression --disable-static --disable-libxcb --disable-sdl2 --disable-xlib --enable-lto --enable-gpl --enable-version3 --enable-shared --enable-gmp --enable-gnutls --enable-chromaprint --enable-libdrm --enable-libass --enable-libfreetype --enable-libfribidi --enable-libfontconfig --enable-libbluray --enable-libmp3lame --enable-libopus --enable-libtheora --enable-libvorbis --enable-libopenmpt --enable-libdav1d --enable-libwebp --enable-libvpx --enable-libx264 --enable-libx265 --enable-libzvbi --enable-libzimg --enable-libfdk-aac --arch=amd64 --enable-libsvtav1 --enable-libshaderc --enable-libplacebo --enable-vulkan --enable-opencl --enable-vaapi --enable-amf --enable-libvpl --enable-ffnvcodec --enable-cuda --enable-cuda-llvm --enable-cuvid --enable-nvdec --enable-nvenc
>   libavutil      58.  2.100 / 58.  2.100
>   libavcodec     60.  3.100 / 60.  3.100
>   libavformat    60.  3.100 / 60.  3.100
>   libavdevice    60.  1.100 / 60.  1.100
>   libavfilter     9.  3.100 /  9.  3.100
>   libswscale      7.  1.100 /  7.  1.100
>   libswresample   4. 10.100 /  4. 10.100
>   libpostproc    57.  1.100 / 57.  1.100
> [mov,mp4,m4a,3gp,3g2,mj2 @ 0x36452170180] Format mov,mp4,m4a,3gp,3g2,mj2 detected only with low score of 1, misdetection possible!
> [mov,mp4,m4a,3gp,3g2,mj2 @ 0x36452170180] moov atom not found
> upload/encoded-video/4a95d843-2054-4ad1-8c09-f59148f960fd/d9/ff/d9ff1cfc-1e92-446a-a8a6-ba1c71624350-MP.mp4: Invalid data found when processing input
> 
>     at ChildProcess.<anonymous> (/usr/src/app/node_modules/fluent-ffmpeg/lib/ffprobe.js:233:22)
>     at ChildProcess.emit (node:events:514:28)
>     at ChildProcess._handle.onexit (node:internal/child_process:294:12)

I think this is valid ffmpeg error message. Probably because of invalid / broken video file.

The problem is that the log no information whatsoever about which file has that issue. You can see "upload/encoded-video/4a95d843-2054-4ad1-8c09-f59148f960fd/d9/ff/d9ff1cfc-1e92-446a-a8a6-ba1c71624350-MP.mp4" in the log, but this is not real filename. This is some Immich internal name, and I have no idea how to find that file in my library.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by SlavikCA** [comment]
Similar message for the broken photo:


> immich_microservices     | [Nest] 7  - 12/18/2023, 2:15:22 AM   ERROR [JobService] Unable to run job handler (thumbnailGeneration/generate-jpeg-thumbnail): Error: Input file contains unsupported image format
> immich_microservices     | [Nest] 7  - 12/18/2023, 2:15:22 AM   ERROR [JobService] Error: Input file contains unsupported image format
> immich_microservices     |     at Sharp.toFile (/usr/src/app/node_modules/sharp/lib/output.js:89:19)
> immich_microservices     |     at MediaRepository.resize (/usr/src/app/dist/infra/repositories/media.repository.js:38:14)
> immich_microservices     |     at MediaService.generateThumbnail (/usr/src/app/dist/domain/media/media.service.js:116:44)
> immich_microservices     |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
> immich_microservices     |     at async MediaService.handleGenerateJpegThumbnail (/usr/src/app/dist/domain/media/media.service.js:103:28)
> immich_microservices     |     at async /usr/src/app/dist/domain/job/job.service.js:113:37
> immich_microservices     |     at async Worker.processJob (/usr/src/app/node_modules/bullmq/dist/cjs/classes/worker.js:387:28)
> immich_microservices     |     at async Worker.retryIfFailed (/usr/src/app/node_modules/bullmq/dist/cjs/classes/worker.js:574:24)
> immich_microservices     | [Nest] 7  - 12/18/2023, 2:15:22 AM   ERROR [JobService] Object:
> immich_microservices     | {
> immich_microservices     |   "id": "6443be3c-74cd-46e1-821a-3560e24d80be",
> immich_microservices     |   "source": "upload"
> immich_microservices     | }


How can I find the photo?

**Reply by hirenshah** [reply]
@SlavikCA For now I am taking that "id" value and running this database query via pgAdmin:
```
SELECT * FROM public.assets
WHERE id='INSERT ID HERE'
ORDER BY id ASC 
```
The originalPath and/or originalFileName have the necessary information in order to find the files in question and delete them.  Once deleted, you would need to follow https://immich.app/docs/guides/remove-offline-files/ to clear up the orphans. 

As far as I can see there isn't a better way at the moment, so hopefully something will get implemented.
